lib/rest-adapter.js

```diff
@@ -143,17 +143,36 @@ RestAdapter.prototype.invoke = function(method, ctorArgs, args, callback) {
   });
 };
 
-RestAdapter.prototype.getRestMethodByName = function(name) {
-  var classes = this.getClasses();
-  for (var i = 0; i < classes.length; i++) {
-    var restClass = classes[i];
-    for (var j = 0; j < restClass.methods.length; j++) {
-      var restMethod = restClass.methods[j];
-      if (restMethod.fullName === name) {
-        return restMethod;
+/**
+ * gets or creates the rest method by name cache map.
+ * @param reload forces the map to refresh if true
+ * @returns {boolean} returns true if the cache was updated
+ */
+RestAdapter.prototype.getOrCreateRestMethodByNameMap = function(reload) {
+  var updated = false;
+  if (!this.restMethodByNameMap || reload) {
+    this.restMethodByNameMap = {};
+    var classes = this.getClasses();
+    for (var i = 0; i < classes.length; i++) {
+      var restClass = classes[i];
+      for (var j = 0; j < restClass.methods.length; j++) {
+        var restMethod = restClass.methods[j];
+        this.restMethodByNameMap[restMethod.fullName] = restMethod;
```

Comment:
The property should be private too. IMO, the name should include "cache" to make the intent more clear, e.g. `this._cachedRestMethods` or `this._cachedRestMethodsByName`.